docs/src/main/tut/docs/rpc/README.md

```diff
@@ -4,6 +4,668 @@ title: RPC
+    - [Plugin Settings](#plugin-settings)
+    - [Generation with protoGen](#generation-with-protogen)
+  - [RPC Service Implementations](#rpc-service-implementations)
+    - [Server](#server)
+    - [Server Runtime](#server-runtime)
+      - [Execution Context](#execution-context)
+      - [Runtime Implicits](#runtime-implicits)
+    - [Server Bootstrap](#server-bootstrap)
+    - [Client](#client)
+    - [Client Runtime](#client-runtime)
+      - [Execution Context](#execution-context-1)
+      - [Runtime Implicits](#runtime-implicits-1)
+    - [Client Program](#client-program)
+  - [frees-rpc annotations](#frees-rpc-annotations)
+  - [Next Steps](#next-steps)
+  - [Comparing HTTP and RPC](#comparing-http-and-rpc)
+- [References](#references)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## What’s frees-rpc
+
+[frees-rpc] brings the ability to combine [RPC] protocols, services and clients in your `Freestyle` program, thanks to [gRPC]. Although it's fully integrated with [gRPC], there are some important differences when defining the protocols, as we’ll see later on, since [frees-rpc] follows the same philosophy that `Freestyle` core, being macro-powered.
+
+## Installation
+
+Add the following dependency to your project's build file.
+
+For Scala `2.11.x` and `2.12.x`:
+
+```scala
+libraryDependencies += "io.frees" %% "frees-rpc" % "0.1.2"
+```
+
+## About gRPC
+
+> [gRPC](https://grpc.io/about/) is a modern open source high performance RPC framework that can run in any environment. It can efficiently connect services in and across data centers with pluggable support for load balancing, tracing, health checking and authentication. It is also applicable in last mile of distributed computing to connect devices, mobile applications and browsers to backend services.
+ 
+In this project, we are focused in the [Java gRPC] implementation.
+
+Now, in the incoming sections let's see how we would implement RPC protocols (Messages and Services) in both *gRPC* and *frees-rpc*.
+
+## Messages and Services
+
+### gRPC
+
+As you might know, [gRPC] uses protocol buffers (a.k.a. *protobuf*) by default:
+
+* As the Interface Definition Language (IDL) for describing both the service interface and the structure of the payload messages. It is possible to use other alternatives if desired. 
+* For serializing/deserializing structure data, similarly as you do with [JSON] data, defining files `.json` extension, with protobuf you have to define proto files with `.proto` as extension.
+
+In the example given in the [gRPC guide], you might have a proto file like this:
+
+```
+message Person {
+  string name = 1;
+  int32 id = 2;
+  bool has_ponycopter = 3;
+}
+```
+
+Then, once you’ve specified your data structures, you can use the protobuf compiler `protoc` to generate data access classes in your preferred language(s) from your proto definition. In the Scala ecosystem, the most widely used is [ScalaPB].
```

Comment:
Are we still using ScalaPB? if not no need to mention it anywhere but in the credits section in terms of inspiration.